### ISSUE TITLE
Would it be possible to create a vst from the ym2608 emulator of the emulator ?

### ISSUE THREAD

**Issue by pikachuk** [AZO234/NP2kai#56]
Hi, i was wondering if it was possible to create a vst from the ym2608 emulation of the emulator, since i can't find any vst online and i wanted to make pc98 styled music with my DAW

since all the source code is already ready i guess it shouldn't be that hard but i wonder if it's doable

**Comment by AZO234** [comment]
If it is the minimum function, I think that it can be realized easily.
· Convert VST and MIDI parameters for OPNA
· Match the MIDI program number and the OPNA preset number
· Input clock and data to fmgen (OPNA engine of NP2kai)
· Output to an arbitrary audio channel

The source of fmgen engine is located at the following site.
http://retropc.net/cisc/m88/download.html
fmgen008.lzh
fmgen simulates FM sound of OPN/OPNA/OPM.

If we can make it well, 
it will be a VST plug-in that will sound FM of PC88/PC98/X68000.

**Comment by pikachuk** [comment]
I didn't find any vst for the ym2608, there's some for the ym2612 of the sega genesis which is an inferior version of the ym2608 but that's all
i have some basic C++ knowledge but i don't know if it requires extensive knowledge of programming midi related stuff because what i'm used to make is basic game engine programming using SDL2

**Comment by pikachuk** [comment]
i'll try to make the vst myself, i'll tell you if i succeed

**Comment by jpcima** [comment]
Hi. Having stumbled by chance on this issue, I guess it's worth mentioning I'm an author of [OPN2 VST](https://github.com/jpcima/ADLplug).
There exists a working attempt to introduce the OPNA emulator in replacement of OPN2 in [recent work](https://github.com/Wohlstand/libOPNMIDI/pull/72). It's for handling the FM part only.

For the most part it works excellently and fast, but it was noted the SSG-EG behaved very different in contrast to Nuked or MAME emulators. (but it can be a mistake made on my programming side)
Do you know how good is the implementation of SSG-EG of Neko Project II known to be?

**Comment by SeleDreams** [comment]
> If it is the minimum function, I think that it can be realized easily. · Convert VST and MIDI parameters for OPNA · Match the MIDI program number and the OPNA preset number · Input clock and data to fmgen (OPNA engine of NP2kai) · Output to an arbitrary audio channel
> 
> The source of fmgen engine is located at the following site. http://retropc.net/cisc/m88/download.html fmgen008.lzh fmgen simulates FM sound of OPN/OPNA/OPM.
> 
> If we can make it well, it will be a VST plug-in that will sound FM of PC88/PC98/X68000.
> 
> Or does MAME-based VST plug-in exist?

I was wondering what's the license of this code ? because all I see at the top is litterally "copyright" it's not under gpl or any open source license i know of

**Comment by StReedTp2** [comment]
> i'll try to make the vst myself, i'll tell you if i succeed

Hello! It's been 7 years now. Do you have any updates ? Have you succeeded in making a YM2608 VST yourself?

**Comment by SeleDreams** [comment]
> > i'll try to make the vst myself, i'll tell you if i succeed
> 
> Hello! It's been 7 years now. Do you have any updates ? Have you succeeded in making a YM2608 VST yourself?

I've been busy with lots of stuff. And since there is the BambooTracker software https://bambootracker.github.io/BambooTracker/
it felt less necessary

**Comment by StReedTp2** [comment]
> > > i'll try to make the vst myself, i'll tell you if i succeed
> > 
> > 
> > Hello! It's been 7 years now. Do you have any updates ? Have you succeeded in making a YM2608 VST yourself?
> 
> I've been busy with lots of stuff. And since there is the BambooTracker software https://bambootracker.github.io/BambooTracker/ it felt less necessary

Ah, I understand. I don't really know how to use a Tracker. I'm looking for a VST mainly because I want to use PC-98 sounds directly in a DAW. It's unfortunate that there still isn't any VST that can do that.

**Comment by wardhale** [comment]
> Ah, I understand. I don't really know how to use a Tracker. I'm looking for a VST mainly because I want to use PC-98 sounds directly in a DAW. It's unfortunate that there still isn't any VST that can do that.

I totally agree here. I'm a traditional composer and I would pay for a real OPNA VST. A tracker is fine, but it's for a totally different type of brain.
